src/devices/Button/ButtonBase.cs

```diff
@@ -0,0 +1,193 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Threading;
+
+namespace Iot.Device.Button
+{
+    /// <summary>
+    /// Base implementation of Button logic.
+    /// Hardware independent. Inherit for specific hardware handling.
+    /// </summary>
+    public class ButtonBase : IDisposable
+    {
+        internal const long DefaultDoublePressTicks = 15000000;
+        internal const long DefaultHoldingMilliseconds = 2000;
+
+        private bool _disposed = false;
+
+        private long _doublePressTicks;
+        private long _holdingMs;
+
+        private ButtonHoldingState _holdingState = ButtonHoldingState.Completed;
+
+        private long _lastPress = DateTime.MinValue.Ticks;
+        private Timer? _holdingTimer;
+
+        /// <summary>
+        /// Delegate for button pressed.
+        /// </summary>
+        /// <param name="sender">The sender object.</param>
+        /// <param name="e">The event data.</param>
+        public delegate void ButtonPressedDelegate(object sender, EventArgs e);
```

Comment:
No need to declare these manually in .Net. Use `Action<T>`

Reply:
@zoekdestep Thanks. In such cases, you can just silently resolve a conversation. This is considered equivalent to "I think I have addressed the issue". Less typing ;-)